src/models/ResourceType.ts

```diff
@@ -15,7 +11,54 @@ export interface Attribute {
   }
 })
 
+@pre<ResourceType>('remove', async function(): Promise<void> {
+  const childTypesCount = await ResourceTypeModel.countDocuments({ parentType: this._id });
+  if (childTypesCount > 0) {
+    return new Promise((resolve, reject) => {
+      reject(new Error(`There are ${childTypesCount} resource types with '${this.name}' as parent. ` +
+        `Type can not be deleted.`));
+    });
+  }
+
+  const instancesCount = await ResourceInstanceModel.countDocuments({ resourceType: this._id });
```

Comment:
Is this really the intended behavoir? Maybe the user wants to delete a resource type including all of its instances.

Reply:
Maybe we can add this functionality later with an additional confirmation, like "This type has 243 instances. Do you really want to delete all instances and the type?". At the moment, I think its easier and safer to allow deletion only if the type is "empty". We of course could think about a route deleting all instances of a type (bulk deleting by property/attribute).